File: lib/features/master_data/domain/entities/tai_khoan_ngan_hang.dart
// ============================================================================
// Domain Layer - Entities
// Based on UC_HKD_TT88_2021 - MD-07: Quản lý danh mục tài khoản ngân hàng
// ============================================================================

import 'package:equatable/equatable.dart';

class TaiKhoanNganHang extends Equatable {
  final String id;
  final String maTaiKhoan;
  final String tenTaiKhoan;
  final String? tenNganHang;
  final String? chiNhanh;
  final String? soTaiKhoan;
  final String? loaiTaiKhoan; // TIET_KIEM / THANH_TOAN / PHAT_TRIEN
  final String? diaChiNganHang;
  final String? soDienThoaiNganHang;
  final String? emailNganHang;
  final String trangThai; // HOAT_DONG / DONG
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const TaiKhoanNganHang({
    required this.id,
    required this.maTaiKhoan,
    required this.tenTaiKhoan,
    this.tenNganHang,
    this.chiNhanh,
    this.soTaiKhoan,
    this.loaiTaiKhoan,
    this.diaChiNganHang,
    this.soDienThoaiNganHang,
    this.emailNganHang,
    this.trangThai = 'HOAT_DONG',
    this.createdAt,
    this.updatedAt,
  });

  TaiKhoanNganHang copyWith({
    String? id,
    String? maTaiKhoan,
    String? tenTaiKhoan,
    String? tenNganHang,
    String? chiNhanh,
    String? soTaiKhoan,
    String? loaiTaiKhoan,
    String? diaChiNganHang,
    String? soDienThoaiNganHang,
    String? emailNganHang,
    String? trangThai,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return TaiKhoanNganHang(
      id: id ?? this.id,
      maTaiKhoan: maTaiKhoan ?? this.maTaiKhoan,
      tenTaiKhoan: tenTaiKhoan ?? this.tenTaiKhoan,
      tenNganHang: tenNganHang ?? this.tenNganHang,
      chiNhanh: chiNhanh ?? this.chiNhanh,
      soTaiKhoan: soTaiKhoan ?? this.soTaiKhoan,
      loaiTaiKhoan: loaiTaiKhoan ?? this.loaiTaiKhoan,
      diaChiNganHang: diaChiNganHang ?? this.diaChiNganHang,
      soDienThoaiNganHang: soDienThoaiNganHang ?? this.soDienThoaiNganHang,
      emailNganHang: emailNganHang ?? this.emailNganHang,
      trangThai: trangThai ?? this.trangThai,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  List<Object?> get props => [
    id,
    maTaiKhoan,
    tenTaiKhoan,
    tenNganHang,
    chiNhanh,
    soTaiKhoan,
    loaiTaiKhoan,
    diaChiNganHang,
    soDienThoaiNganHang,
    emailNganHang,
    trangThai,
    createdAt,
    updatedAt,
  ];
}

File: HkdAccounting.Domain/Entities/TaiKhoanNganHang.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HkdAccounting.Domain.Entities
{
    /// <summary>
    /// Tài khoản ngân hàng
    /// </summary>
    public class TaiKhoanNganHang
    {
        /// <summary>
        /// Khóa chính
        /// </summary>
        public long Id { get; set; }

        /// <summary>
        /// Mã tài khoản
        /// </summary>
        [Required(ErrorMessage = "Mã tài khoản là bắt buộc")]
        [MaxLength(20, ErrorMessage = "Mã tài khoản không được vượt quá 20 ký tự")]
        public string MaTaiKhoan { get; set; }

        /// <summary>
        /// Số tài khoản
        /// </summary>
        [Required(ErrorMessage = "Số tài khoản là bắt buộc")]
        [MaxLength(50, ErrorMessage = "Số tài khoản không được vượt quá 50 ký tự")]
        public string SoTaiKhoan { get; set; }

        /// <summary>
        /// Tên ngân hàng
        /// </summary>
        [Required(ErrorMessage = "Tên ngân hàng là bắt buộc")]
        [MaxLength(255, ErrorMessage = "Tên ngân hàng không được vượt quá 255 ký tự")]
        public string TenNganHang { get; set; }

        /// <summary>
        /// Chi nhánh
        /// </summary>
        [MaxLength(255, ErrorMessage = "Chi nhánh không được vượt quá 255 ký tự")]
        public string ChiNhanh { get; set; }

        /// <summary>
        /// Số dư hiện tại
        /// </summary>
        [Column(TypeName = "decimal(18,2)")]
        public decimal SoDuHienTai { get; set; } = 0;

        /// <summary>
        /// Khóa ngoại tới HkdInfo
        /// </summary>
        public long HkdInfoId { get; set; }

        /// <summary>
        /// Trạng thái
        /// </summary>
        [MaxLength(20, ErrorMessage = "Trạng thái không được vượt quá 20 ký tự")]
        public string TrangThai { get; set; } = "DANG_SU_DUNG";

        /// <summary>
        /// Ngày tạo bản ghi
        /// </summary>
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        /// <summary>
        /// Ngày cập nhật bản ghi
        /// </summary>
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public virtual HkdInfo HkdInfo { get; set; }
    }
}
